### ISSUE TITLE
Incompatibility Issue between node-celery-ts (client) and node-celery (worker) due to Different Body Encoding Expectations

### ISSUE THREAD

**Issue by anasch07** [node-celery-ts/node-celery-ts#93]
Hello everyone,

I have been encountering an issue while trying to use `node-celery-ts` as a client and `node-celery` as a worker. The problem arises due to a mismatch in the expected body encoding between the two.

In my setup, the `node-celery-ts` client sends a message with a `body_encoding` property set to "utf-8". However, the `node-celery` worker expects the `body_encoding` to be "base64". As a result, the worker throws an error "unsupported body encoding utf-8".

Here is a snippet of the worker code where the error is thrown:

```javascript
// now supports only base64 of body_encoding
if (rawMsg.properties.body_encoding !== "base64") {
    throw new Error(`queue ${queue} item: unsupported body encoding ${rawMsg.properties.body_encoding}`);
}
```

Unfortunately, the `node-celery-ts` package does not seem to provide an option to change the `body_encoding` property.

I have considered several workarounds, such as modifying the worker code to accept "utf-8" encoding, using a different client or worker, or modifying the client code to send "base64" encoding. However, these are not ideal solutions.

Ideally, the `node-celery-ts` and `node-celery` packages should be compatible with each other. This might require changes to the packages themselves to ensure they can work together regardless of the body encoding.

Has anyone else encountered this issue? Any suggestions or solutions would be greatly appreciated.

Thank you!

**Comment by cwill747** [comment]
This repo is fairly unmaintained at this point because I don’t use it at work anymore, but let me see if I can clear out this and some of the other issues in the next couple of weeks.On Apr 23, 2024, at 6:44 PM, Anas Chaibi ***@***.***> wrote:﻿
Hello everyone,
I have been encountering an issue while trying to use node-celery-ts as a client and node-celery as a worker. The problem arises due to a mismatch in the expected body encoding between the two.
In my setup, the node-celery-ts client sends a message with a body_encoding property set to "utf-8". However, the node-celery worker expects the body_encoding to be "base64". As a result, the worker throws an error "unsupported body encoding utf-8".
Here is a snippet of the worker code where the error is thrown:
// now supports only base64 of body_encoding
if (rawMsg.properties.body_encoding !== "base64") {
    throw new Error(`queue ${queue} item: unsupported body encoding ${rawMsg.properties.body_encoding}`);
}
Unfortunately, the node-celery-ts package does not seem to provide an option to change the body_encoding property.
I have considered several workarounds, such as modifying the worker code to accept "utf-8" encoding, using a different client or worker, or modifying the client code to send "base64" encoding. However, these are not ideal solutions.
Ideally, the node-celery-ts and node-celery packages should be compatible with each other. This might require changes to the packages themselves to ensure they can work together regardless of the body encoding.
Has anyone else encountered this issue? Any suggestions or solutions would be greatly appreciated.
Thank you!

**Comment by anasch07** [comment]
I think there is an opened pr to fix this issue !

**Comment by cwill747** [comment]
Yep! I need to make sure everything builds first and then I’ll get a release out when I can. On Apr 23, 2024, at 6:54 PM, Anas Chaibi ***@***.***> wrote:﻿
I think there is an opened pr to fix this issue !

**Comment by anasch07** [comment]
Much appreciated !
